Samples/NetworkMediaStreamer/RTSPServer/src/SocketWrapper.cpp

```diff
@@ -0,0 +1,304 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for more information
+
+#include <pch.h>
+
+CSocketWrapper::CSocketWrapper(SOCKET connectedSocket, winrt::array_view<PCCERT_CONTEXT> aCertContext /*= empty vector*/)
+    : m_bIsSecure(!aCertContext.empty())
+    , m_socket(connectedSocket)
+    , m_hCred({ 0 })
+    , m_hCtxt({ 0,0 })
+    , m_aCertContext(aCertContext)
+    , m_secPkgContextStrmSizes({ 0 })
+    , m_bufSz(0)
+    , m_readEvent(WSA_INVALID_EVENT)
+    , m_callBackHandle(nullptr)
+    , m_bIsAuthenticated(false)
+{
+    if (m_bIsSecure)
+    {
+
+        InitilizeSecurity();
+
+        winrt::check_hresult(QueryContextAttributes(

```

Comment:
> winrt::check_hresul [](start = 7, length = 20)

throwing from within a constructor without catch, beware

Reply:
there is a try catch across the call to the constructor in RTSPServer.cpp 

---
In reply to: [493000186](https://github.com/microsoft/Windows-Camera/pull/13#discussion_r493000186) [](ancestors = 493000186)